docs: expand self-hosting instructions (#4223)
KaTeX/KaTeX: docs/browser.md

Reviewer: Should this come earlier? I think it might be more common (and better advice) than self copying...

to provide a fallback for older browsers that do not support ES modules.

## Download & Host Things Yourself
If you do not want to rely on a CDN, you can serve KaTeX from your own
infrastructure. There are two ways to get a release:

### Option 1: Pre-built release from GitHub

Each release ships pre-built `katex.tar.gz` and `katex.zip` archives.
Download one from the [KaTeX releases page](https://github.com/KaTeX/KaTeX/releases) —

> look under **Assets** for `katex.tar.gz` or `katex.zip`, **not** the

Reviewer: ```suggestion
look under **Assets** for `katex.tar.gz` or `katex.zip`, *not* the
```

```
katex/
├── README.md
├── katex.js              # unminified UMD build
├── katex.min.js          # minified UMD build (recommended for production)
├── katex.mjs             # ES module build
├── katex.css             # stylesheet (font-display: block)
├── katex.min.css         # minified stylesheet
├── katex-swap.css        # stylesheet variant (font-display: swap)
├── katex-swap.min.css
├── contrib/              # optional extensions (each shipped as .js, .min.js, and .mjs)
│   ├── auto-render.js
│   ├── auto-render.min.js
│   ├── auto-render.mjs
│   ├── copy-tex.js
│   ├── copy-tex.min.js
│   ├── copy-tex.mjs
│   ├── mathtex-script-type.js
│   ├── mathtex-script-type.min.js
│   ├── mathtex-script-type.mjs
│   ├── mhchem.js
│   ├── mhchem.min.js
│   ├── mhchem.mjs
│   ├── render-a11y-string.js
│   ├── render-a11y-string.min.js
│   └── render-a11y-string.mjs
└── fonts/                # WOFF2/WOFF/TTF font files (required)
```

### Option 2: npm

```bash
npm install katex
# or
yarn add katex
```

After installing, the files you want for hosting live under
`node_modules/katex/dist/` — same layout as the `katex/` folder above.

> > The npm package also ships the un-built source (`node_modules/katex/src/`,

Reviewer: ```suggestion
> The npm package also ships the unbuilt TypeScript source (`node_modules/katex/src/`,
```

HTML will load KaTeX with auto-rendering enabled:

```html
<!DOCTYPE html>
<html>
  <head>
    <link rel="stylesheet" href="/katex/katex.min.css">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="/katex/katex.min.js"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="/katex/contrib/auto-render.min.js"
        onload="renderMathInElement(document.body);"></script>
  </head>
  <body>
    ...
  </body>
</html>
```

Adjust the `/katex/` prefix to match wherever you placed the files. If
you only need to render math via the JavaScript API (and not the
auto-render extension), you can omit the `auto-render.min.js` script.
Swap `katex.min.css`, `katex.min.js`, and `contrib/auto-render.min.js`
for the unminified `katex.css`, `katex.js`, and `contrib/auto-render.js`
if you prefer readable files for debugging.

### Keeping your copy up to date

Every release publishes the same file names at the same paths, so
updating is a drop-in replace. To update:

1. Get the new release files:
   - **Archive:** download a fresh `katex.tar.gz` (or `katex.zip`) and
     extract it.

>    - **npm:** run `npm update katex` (or `yarn upgrade katex`). The new

Reviewer: ```suggestion
   - **npm:** run `npm install katex@latest` (or `yarn upgrade katex@latest`). The new
```

npm upgrade does not bump to latest

     files appear in `node_modules/katex/dist/`.
2. Replace the `katex/` directory on your web server with the new
   files — extracted from the archive, or copied from
   `node_modules/katex/dist/`. Keep the same folder name so existing

>    HTML keeps working. **Note:** `npm update katex` only refreshes the

Reviewer: This Note seems a bit hand-holdy to me. Doesn't step 2 clearly come after step 1?
